src/core/tests/type_prop/abs.cpp

```diff
@@ -4,8 +4,96 @@
 
 #include "openvino/op/abs.hpp"
 
+#include <gtest/gtest.h>
+
+#include "openvino/core/validation_util.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/shape_of.hpp"
 #include "unary_ops.hpp"
 
 using Type = ::testing::Types<ov::op::v0::Abs>;
 
 INSTANTIATE_TYPED_TEST_SUITE_P(type_prop_abs, UnaryOperator, Type);
+
+struct bound {
+    int64_t lower, upper, expected_lower, expected_upper;
+    std::shared_ptr<ov::Symbol> in_symbol;
+};
+
+using AbsTestParams = std::vector<bound>;
+
+class AbsTest : public testing::TestWithParam<AbsTestParams> {};
+
+namespace {
+std::shared_ptr<ov::Node> construct_graph_with_partial_value(const AbsTestParams& params) {
+    auto shape = std::vector<ov::Dimension>();
+    auto subtrahend = std::vector<int64_t>();
+
+    for (const auto& i_bound : params) {
+        if (i_bound.lower >= 0) {
+            shape.emplace_back(i_bound.lower, i_bound.upper);
+            subtrahend.push_back(0);
+        } else {
+            shape.emplace_back(0, (i_bound.upper - i_bound.lower));
+            subtrahend.push_back(i_bound.lower);
+        }
+    }
+
+    auto parameter = std::make_shared<ov::op::v0::Parameter>(ov::element::dynamic, shape);
+    auto shape_of = std::make_shared<ov::op::v3::ShapeOf>(parameter);
+    auto subtract = std::make_shared<ov::op::v1::Add>(
+        shape_of,
+        ov::op::v0::Constant::create(element::i64, ov::Shape{subtrahend.size()}, subtrahend));
+    return subtract;
+}
+}  // namespace
+
+TEST_P(AbsTest, type_prop_abs) {
```

Comment:
```suggestion
TEST_P(TypePropAbsV0Test , bounds) {
```
`bounds` can be replaced by more meaningful name.